xlive/H2MOD_Network.cpp

```diff
@@ -529,12 +532,20 @@ int __cdecl serializeChatPacket(void* a1, int a2, int a3) {
 	}
 	int result = getDataEncodeStringMethod()(a1, (int)"text", a3 + 160, 121);
 
+
 	if (network->networkCommand != NULL && network->networkCommand[0] != '\0') {
-		char* commandToWrite = (char*)(a3 + (CHAT_PACKET_SIZE - 121));
-		memset(commandToWrite, 0, 121);
-		memcpy(commandToWrite, network->networkCommand, strlen(network->networkCommand));
-		//TODO: if text is precense, don't send over the command, as this call stack is from someone sending text
-		result = getDataEncodeStringMethod()(a1, (int)"command", (int)commandToWrite, 121);
+		std::string encoded_message = base64_encode(network->networkCommand, network->command_size);
+		INT32 packet_size = encoded_message.size();
+		if (LOG_CHECK(packet_size <= CHAT_PACKET_EXTEND_SIZE))
+		{
```

Comment:
if you're gonna put a open brace on the else below, do same for this if, please consistent